snooty/postprocess.py

```diff
@@ -568,10 +589,21 @@ def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
             }
             self.selectors[tabset_name].append(tabs)
 
+    def exit_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
+        if not isinstance(node, n.Directive):
+            return
+
+        popped_item = self.stack.pop()
+        self.scanned_pattern.append(popped_item)
```

Comment:
The order in which nodes are entered and exited is fairly similar to the order of the stack itself. Would it make sense to omit the extra need for a separate stack and instead appending to `self.scanned_pattern` on node enter, and popping on node exit directly? Or is there a specific need for both `self.stack` and `self.scanned_pattern`?

Reply:
@rayangler my initial though was it was easier to read. But essentially I think it would be fair or possible to omit the need for the stack and just use `self.scanned_pattern`

Reply:
@rayangler let me know your thoughts. I can refactor to reduce to one stack if it makes more sense.